lib/index.js

```diff
@@ -112,11 +112,14 @@ const os = require('os'),
                 $self: [selfModelName, selfId]
             },
             cursor: data.cursor,
-            context: {
-                ref: data.cursor.ref,
-                httpRequestId: data.cursor.httpRequestId,
-                scriptId: data.cursor.scriptId,
-                execution: data.cursor.execution
+            args: data.cursor && {
+                [EXECUTION_TYPES.request]: data.cursor.httpRequestId && {
+                    $self: [SELF_MODELS.requests, data.cursor.httpRequestId]
+                },
+                event: data.cursor.scriptId, // events don't have a self-reference model
+                [EXECUTION_TYPES.script]: data.cursor.execution && {
```

Comment:
The naming `EXECUTION_TYPES` is confusing as to why an execution is taking other executions as `args`